### PR TITLE
Improve HmacTestVector test

### PR DESCRIPTION
**Pull request by blukat29**
The size of a HMAC key is recommended to be at least as long as its output size. RFC 2104 mentions:

> The key for HMAC can be of any length (keys longer than B bytes are
  first hashed using H).  However, less than L bytes is strongly
  discouraged as it would decrease the security strength of the
  function.

Because of this, C_SignInit may return CKR_KEY_SIZE_RANGE if too short key was given. This patch will
- Skip a test case when CKR_KEY_SIZE_RANGE is returned,
- Add HMAC-SHA2 test vectors where the key size is longer than the HMAC output sizes.

### CODE REVIEW

File: hmac.cc [hunk 1]
```diff
@@ -61,17 +61,23 @@ map<string, vector<TestData> > kTestVectors = {
    {{"0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b", "4869205468657265",
      "b0344c61d8db38535ca8afceaf0bf12b881dc200c9833da726e9376c2e32cff7"},
     {"4a656665", "7768617420646f2079612077616e7420666f72206e6f7468696e673f",
-     "5bdcc146bf60754e6a042426089575c75a003f089d2739839dec58b964ec3843"}, }},
+     "5bdcc146bf60754e6a042426089575c75a003f089d2739839dec58b964ec3843"},
+    {"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", "54657374205573696e67204c6172676572205468616e20426c6f636b2d53697a65204b6579202d2048617368204b6579204669727374",
+     "60e431591ee0b67f0d8a26aacbf5b77f8e0bc6213728c5140546040f0ee37f54"}, }},
```
Comment (daviddrysdale) [inline hmac.cc:66]: Thanks, these seem like useful extra test cases to include.

However, they don't really belong in this table &ndash; they're not test vectors from the specific RFCs mentioned.  Could you pull them out into a separate `HmacKeyTooShort` test case please?

Comment (blukat29) [reply hmac.cc:66]: They are actually from RFC 4231 - test case 6

Comment (daviddrysdale) [reply hmac.cc:66]: Ah, sorry, I should have gone back and checked.

In that case, maybe add a `bool` field to `TestData` to indicate whether a `CKR_KEY_SIZE_RANGE` error is allowed?  I think it makes sense to fail the test when that error shouldn't happen.

Comment (blukat29) [reply hmac.cc:66]: That makes sense. I will add a commit.

File: hmac.cc [hunk 2]
```diff
@@ -42,42 +42,46 @@ struct TestData {
...
 
 map<string, vector<TestData> > kTestVectors = {
   // Test vectors from RFC 2202.
   {"MD5-HMAC",
    {{"0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b", "4869205468657265",
-     "9294727a3638bb1c13f48ef8158bfc9d"},
+     "9294727a3638bb1c13f48ef8158bfc9d", false},
     {"4a656665", "7768617420646f2079612077616e7420666f72206e6f7468696e673f",
-     "750c783e6ab0b503eaa86e310a5db738"}}},
+     "750c783e6ab0b503eaa86e310a5db738", true}}},
```
Comment (daviddrysdale) [inline hmac.cc]: Hmm, I wonder now whether this should be auto-determined (i.e have `bool short_key = (testcase.key.size() < testdata.data.size());` in the test body) &ndash; wdyt?  (Sorry, should have thought of that before.)

Comment (blukat29) [reply hmac.cc]: That looks good to me, I should have though about that..
Anyway I'll fix that one.

**Review by daviddrysdale — APPROVED** [review]
Nice, thanks.

**Pull request merged**